packages/inter-protocol/src/auction/auctionBook.js

```diff
@@ -528,14 +663,21 @@ export const prepareAuctionBook = (baggage, zcf) => {
         },
         getSeats() {
           const { collateralSeat, currencySeat } = this.state;
-          this.state.totalProceedsGoal = null;
           return { collateralSeat, currencySeat };
         },
         exitAllSeats() {
           const { priceBook, scaledBidBook } = this.state;
           priceBook.exitAllSeats();
           scaledBidBook.exitAllSeats();
         },
+        endAuction() {
+          const { state } = this;
+
+          state.startCollateral = null;
+          state.lockedPriceForRound = null;
+          state.curAuctionPrice = null;
+          state.remainingProceedsGoal = null;
```

Comment:
that the nullness of these four values has to be synchronized worries me. if they're really non or not at the same time, wouldn't it be better to bundle them in an object? then "between auctions" that single object ref is null and otherwise it's populated in a verifiable "during auction" shape

Reply:
They're not as parallel as they looked. After looking more closely, 
 * startCollateral will never be null. it's reset to empty at the end of an auction, and increases when `addAssets()` is called
 * `startProceedsGoal` is reset at the end of an auction. it increases when `addAssets()` is called with a goal.
 * `lockedPriceForRound` is assigned a value to lock the price, and reset to null at the end of the auction.
 * `curAuctionPrice` is assigned a value at the beginning of each descending step, and reset at the end of the auction.
 * `remainingProceedsGoal` is assigned a value at the beginning of the auction (if there's a goal), reduced on every trade, and reset at the end of the auction.